node/docs/as-thrift.md

```diff
@@ -0,0 +1,159 @@
+# as/thrift
+
+The following is documentation on how to use the `as/thrift` implementation
+to handle encoding and decoding for you
+
+## Thrift example
+
+`thrift/service.thrift`:
+
+```thrift
+struct EchoResult {
+    1: string value
+}
+
+service Echo  {
+    EchoResult echo(
+        1: string value
+    )
+}
+```
+
+`server.js:`
+
+```js
+var TChannelThrift = require('tchannel/as/thrift');
+var TChannel = require('tchannel');
+var fs = require('fs');
+var path = require('path');
+
+var server = TChannel({
+    serviceName: 'server'
+});
+var client = TChannel();
+var tchannelThrift = TChannelThrift({
+    channel: client,
+    source: fs.readFileSync(
+        path.join(__dirname, 'thrift', 'service.thrift'), 'utf8'
+    )
+});
+
+var context = {};
+
+tchannelThrift.register(server, 'echo', context, echo);
+function echo(context, req, head, body, callback) {
+    callback(null, {
+        ok: true,
+        head: head,
+        body: body
+    });
+}
+
+server.listen(4040, '127.0.0.1', onListening);
+
+function onListening() {
+    tchannelThrift.request({
+        serviceName: 'server',
+        host: '127.0.0.1:4040'
+    }).send('Echo::echo', {
+        someHeader: 'headerValue'
+    }, {
+        value: 'some-string'
+    }, onResponse);
+
+    function onResponse(err, resp) {
+        if (err) {
+            console.log('got error', err);
+        } else {
+            console.log('got resp', resp);
+        }
+
+        server.close();
+    }
+}
+```
+
+### `var tchannelThrift = TChannelThrift(opts)`
+
+`TChannelThrift` returns a `tchannelThrift` interface with a 
+`.request()`, `.send()` and `.register()` method used to 
+send call requests and register call request handlers
+
+It can be passed options.
+
+ - required `opts.source` The thrift idl as a string
+ - `opts.bossMode` bossMode defaults to false. When bossMode
```

Comment:
Not sure if we really need this bossMode. If it is critical we check it; otherwise, we warn it. From what I see in the code, warning might be the best choice.